bucket/downloader.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package bucket
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type blockReader interface {
+	Read(blockID flow.Identifier) ([]byte, error)
+}
```

Comment:
I'm not sure why this interface is needed here. The `Downloader` already implements the interface itself, why would you need to wrap it around the interface instead of using the `BlockReader` as the downloader directly?